src/benchmarklib/tpch/tpch_table_generator.cpp

```diff
@@ -295,20 +295,6 @@ std::unordered_map<std::string, BenchmarkTableInfo> TPCHTableGenerator::generate
   return table_info_by_name;
 }
 
-AbstractTableGenerator::IndexesByTable TPCHTableGenerator::_indexes_by_table() const {
-  return {
-      {"part", {{"p_partkey"}}},
-      {"supplier", {{"s_suppkey"}, {"s_nationkey"}}},
-      // TODO(anyone): multi-column indexes are currently not used by the index scan rule and the translator
-      {"partsupp", {{"ps_partkey", "ps_suppkey"}, {"ps_suppkey"}}},  // ps_partkey is subset of {ps_partkey, ps_suppkey}
-      {"customer", {{"c_custkey"}, {"c_nationkey"}}},
-      {"orders", {{"o_orderkey"}, {"o_custkey"}}},
-      {"lineitem", {{"l_orderkey", "l_linenumber"}, {"l_partkey", "l_suppkey"}}},
-      {"nation", {{"n_nationkey"}, {"n_regionkey"}}},
-      {"region", {{"r_regionkey"}}},
-  };
-}
```

Comment:
>[Bouncner](https://github.com/Bouncner) [3 days ago](https://github.com/hyrise/hyrise/pull/2448/files#r1011761942)
>Why remove?
First, we don't support multi-column table indexes. Thus, if one uses the benchmark flag to create table indexes, the index creation procedure will fail (see https://github.com/hyrise/hyrise/blob/0af6eca05bd887f6f68d0aca11f4c155ae66b0ac/src/benchmarklib/abstract_table_generator.cpp#L390) because the above index specifications contain multi-column indexes.

Second, Hyrise currently does not create physical query plans consisting of operators leveraging existing indexes (please correct me if I'm wrong). Thus, even if we would actually create indexes using the corresponding benchmark flag for table/chunk indexes, they would not be used.

For that reason, we suggest removing them until the optimizer actually generates query plans with operators leveraging existing indexes.

Reply:
I am somehow not a fan of this solution. First, there is nothing that forces us to index all the columns of the primary keys. We could only add the first PK column for every table. Second, there are index scans we automatically create and execute in the optimizer (index scan rule). I don't know if any of our analytical benchmark would use them (the selectivity needs to be rather small and OLAP usually does not filter on PK columns).

I don't see a good reason for removing it now. Keeping it forces us to look at it and does not remove already existing functionality.

Reply:
I haven't checked it recently, but I think that the optimizer does not create query plans with the index scan for the TPCH workload. Nevertheless, I'm okay with changing the index specifications here so that only single-column indexes are created and keeping the index specifications here.

Reply:
As far as I know, there are no scans on (first) PK columns in the TPC-H benchmark.